nav2_behavior_tree/include/nav2_behavior_tree/bt_action_server_impl.hpp

```diff
@@ -246,6 +251,28 @@ void BtActionServer<ActionT>::executeCallback()
   }
 }
 
+template<class ActionT>
+void BtActionServer<ActionT>::populateErrorCode(typename std::shared_ptr<typename
+    ActionT::Result> result)
+{
+    // Grab error codes from blackboard
+    std::vector<int> error_codes;
+
+    // 0-99 error codes for follow path
+    error_codes.push_back(blackboard_->get<int>("follow_path_error_code"));
+
+  // 100-199 error codes for compute path to pose or compute path through poses
+  int compute_path_error_code;
+  if (blackboard_->get<int>("compute_path_to_pose_error_code", compute_path_error_code)){
```

Comment:
Its not necessarily true that just because this doesn't exist, the other does. There could be a misconfiguration.

Reply:
For example as I mention in a comment below if the BT port name doesn't match the param name

Reply:
Also use the parameterized names